### PR TITLE
Feat: Add Velero Mixin

### PR DESCRIPTION
**Pull request by BominRahmani**
The following PR will serve as the base PR for which every other Velero mixin PR will merge into. The base PR contains only the ReadMe

### CODE REVIEW

File: velero-2-mixin/alerts.libsonnet [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+              summary: 'Velero backup failures detected.',
+              description: |||
+                Backup failures detected on {{ $labels.instance }}. This could lead to data loss or inability to recover in case of a disaster.
+              ||| % this.config,
+            },
+          },
+          {
+            alert: 'VeleroHighBackupDuration',
+            expr: |||
+              histogram_quantile(0.5, sum(rate(velero_backup_duration_seconds_bucket{%(filteringSelector)s}[5m])) by (le)) > %(alertsHighBackupDuration)s * avg_over_time(histogram_quantile(0.5, sum(rate(velero_backup_duration_seconds_bucket{%(filteringSelector)s}[5m])) by (le))[48h:5m])
```
Comment (v-zhuravlev) [inline velero-2-mixin/alerts.libsonnet:28]: Great alert overall! Have you got any chance to test its behavior? I think it might be prone to produce false positives:
1) aggregating different backup jobs together (by le only). Perhaps, this should be aggregated by backup jobs to have better accuracy?

File: velero-2-mixin/alerts.libsonnet [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+              up{%(filteringSelector)s} != %(alertsVeleroUpStatus)s
+            ||| % this.config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Velero is down.',
+              description: |||
+                Cannot find any metrics related to Velero on {{ $labels.instance }}. This indicates that Velero is not running or the Grafana agent is not properly configured.
```
Comment (v-zhuravlev) [inline velero-2-mixin/alerts.libsonnet:69]:
```suggestion
                Cannot find any metrics related to Velero on {{ $labels.instance }}. This indicates that Velero is not running or the Grafana agent is not properly configured.
```
Scraping agents could be different: Grafana Agent/ Alloy/ Prometheus etc...

Comment (v-zhuravlev) [reply velero-2-mixin/alerts.libsonnet:69]: Overall, I would rephrase this alert to be more specific. It is not necessary indicates that Velero is down. It could be just that agent can't scrape metric for various reasons: like metrics endpoint is not configured or auth issues etc...

File: velero-2-mixin/variables.libsonnet [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+// Generates chained variables to use on on all dashboards
+{
+  new(this, varMetric):
+    {
+      local filteringSelector = this.config.filteringSelector,
+      local groupLabels = this.config.groupLabels,
+      local instanceLabels = this.config.instanceLabels,
+      local scheduleLabel = 'schedule',
+      local groupVarMetric = 'velero_backup_attempt_total',
+      local querySelectorVar = ['job', 'cluster', 'instance', 'schedule'],
```
Comment (v-zhuravlev) [inline velero-2-mixin/variables.libsonnet:15]: shouldn't those be a combination of groupLabels, instanceLabels?

File: velero-2-mixin/variables.libsonnet
Comment (v-zhuravlev) [inline velero-2-mixin/variables.libsonnet:1]: You can drop this file altogether and import it inside main.libsonnet from commonlib now:
https://github.com/grafana/jsonnet-libs/blob/master/common-lib/common/variables/variables.libsonnet

File: velero-2-mixin/variables.libsonnet (hunk 3 above)
Comment (BominRahmani) [reply velero-2-mixin/variables.libsonnet]: Yea, I've updated it now so its a combination of groupLabels, instanceLabels, and then the schedule label. Let me know what you think.

File: velero-2-mixin/variables.libsonnet
Comment (BominRahmani) [reply velero-2-mixin/variables.libsonnet:1]: Since i've had some further modifications done in this variables file, i am going to keep it.

File: velero-2-mixin/alerts.libsonnet (hunk 1 above)
Comment (BominRahmani) [reply velero-2-mixin/alerts.libsonnet:28]: I've set it up and have been keeping an eye on it. I havent seen it produce any false positives yet, but that maybe just luck. I can try doing aggregations by backup jobs as well though. I do not feel strongly about it either way. Let me know what you think or maybe @mshahzeb can weigh in on what he thinks as well

File: velero-2-mixin/mixin.libsonnet [hunk 4]
```diff
@@ -0,0 +1,19 @@
+local velerolib = import './main.libsonnet';
+
+local velero =
+  velerolib.new()
+  + velerolib.withConfigMixin(
+    {
+      filteringSelector: 'job=~"integrations/velero"',
```
Comment (mshahzeb) [inline velero-2-mixin/mixin.libsonnet:7]: Since this is a public mixin in jsonnet-lib. I am wondering if we should be hardcoding the selector `integrations/velerio` here?

File: velero-2-mixin/Makefile [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+dashboards_out: mixin.libsonnet config.libsonnet dashboards.libsonnet
+	@mkdir -p dashboards_out
+	mixtool generate dashboards mixin.libsonnet -d dashboards_out
+
+prometheus_alerts.yaml: mixin.libsonnet alerts.libsonnet
+	mixtool generate alerts mixin.libsonnet -a prometheus_alerts.yaml
+
+.PHONY: clean
+clean:
+	rm -rf dashboards_out prometheus_alerts.yaml
```
Comment (mshahzeb) [inline velero-2-mixin/Makefile:34]: Missing newline

**Review by mshahzeb — CHANGES_REQUESTED** [review]

File: velero-2-mixin/mixin.libsonnet (hunk 4 above)
Comment (BominRahmani) [reply velero-2-mixin/mixin.libsonnet:7]: Yea i think this is fine, it is how its been done historically.

Comment (v-zhuravlev) [reply velero-2-mixin/mixin.libsonnet:7]: Yes, we didn't pay attention to this. Overall, I think putting less strict filter on the mixin level here would be better, then overriding in integrations.

File: velero-2-mixin/alerts.libsonnet (hunk 1 above)
Comment (v-zhuravlev) [reply velero-2-mixin/alerts.libsonnet:28]: If you tend to keep it I suggest two things:
- Split alerts by backup jobs separately so previous day stats are more accurate
- Try add `keep_firing_for: 24h` So alert don't get resolved after 5 minutes after alert is triggered. with keep_firing_for that long I expect it only gets resolved if next day backup is still having issues.

**Review by mshahzeb — APPROVED** [review]
LGTM! Thanks!

**Review by v-zhuravlev — APPROVED** [review]
